managers/bosh-manager/DirectorService.js

```diff
@@ -0,0 +1,1112 @@
+const bosh = require('../../data-access-layer/bosh');
+const Agent = require('../../data-access-layer/service-agent');
+const NetworkSegmentIndex = bosh.NetworkSegmentIndex;
+const ServiceBindingAlreadyExists = errors.ServiceBindingAlreadyExists;
+const backupStore = require('../../data-access-layer/iaas').backupStore;
+const boshOperationQueue = bosh.BoshOperationQueue;
+const ServiceInstanceAlreadyExists = errors.ServiceInstanceAlreadyExists;
+const ServiceInstanceNotOperational = errors.ServiceInstanceNotOperational;
+const FeatureNotSupportedByAnyAgent = errors.FeatureNotSupportedByAnyAgent;
+const ServiceBindingNotFound = errors.ServiceBindingNotFound;
+const DeploymentDelayed = errors.DeploymentDelayed;
+const BaseDirectorService = require('../BaseDirectorService');
+const cf = require('../../data-access-layer/cf');
+const cloudController = cf.cloudController;
+const serviceFabrikClient = cf.serviceFabrikClient;
+const Header = bosh.manifest.Header;
+const Addons = bosh.manifest.Addons;
+const EvaluationContext = bosh.EvaluationContext;
+const BadRequest = errors.BadRequest;
+const BasePlatformManager = require('../../broker/lib/fabrik/BasePlatformManager');
+
+
+class DirectorService extends BaseDirectorService {
+  constructor(guid, plan) {
+    super(plan);
+    this.guid = guid;
+    this.plan = plan;
+    this.cloudController = cloudController;
+    this.serviceFabrikClient = serviceFabrikClient;
+    this.director = bosh.director;
+    this.networkSegmentIndex = undefined;
+    this.platformManager = undefined;
+    this.backupStore = backupStore;
+    this.agent = new Agent(this.settings.agent);
+    this.prefix = CONST.SERVICE_FABRIK_PREFIX;
+  }
+
+  assignPlatformManager(platformManager) {
+    this.platformManager = platformManager;
+  }
+
+  get platformContext() {
+    return Promise.try(() => this.networkSegmentIndex ? this.deploymentName : this.director.getDeploymentNameForInstanceId(this.guid))
+      .then(deploymentName => this.director.getDeploymentProperty(deploymentName, CONST.PLATFORM_CONTEXT_KEY))
+      .then(context => JSON.parse(context))
+      .catch(NotFound, () => {
+        /* Following is to handle existing deployments. 
+           For them platform-context is not saved in deployment property. Defaults to CF.
+         */
+        logger.warn(`Deployment property '${CONST.PLATFORM_CONTEXT_KEY}' not found for instance '${this.guid}'.\ 
+        Setting default platform as '${CONST.PLATFORM.CF}'`);
+
+        const context = {
+          platform: CONST.PLATFORM.CF
+        };
+        return context;
+      });
+  }
+
+  static get prefix() {
+    return _
+      .reduce(config.directors,
+        (prefix, director) => director.primary === true ? director.prefix : prefix,
+        null) || super.prefix;
+  }
+
+  get deploymentName() {
+    return this.getDirectorDeploymentName(this.guid, this.networkSegmentIndex);
+  }
+
+  getDirectorDeploymentName(guid, networkSegmentIndex) {
```

Comment:
why is this function named `getDirectorDeploymentName` and not just `getDeploymentName`?

Reply:
adding it in TODO